src/summary.rs

```diff
@@ -0,0 +1,86 @@
+use std::collections::BTreeMap;
+
+use url::Url;
+
+use crate::logger::gha_output;
+
+#[derive(serde::Serialize)]
+pub struct Summary {
```

Comment:
Comments on the field, what they are supposed to represent?

Reply:
As this is serialized no one will see these comments except when looking into the code. Improving the key names is possible, adding comments is pointless.

Reply:
Why are comments pointless here? "..except when looking into the code." -> That's exactly what I am doing and also the reason I ask for comments. The fields names might be clear for you but not for others. Some of them are ambiguous: `pub changed_amount: usize` Amount in what? Emojis? Lines? Bytes?

Yes I am a bit pedantic here but comments usually don't harm and I consider them part of the engineering process during software development!

Reply:
When this is not clear by looking on the serialized JSON it's not good enough. People will use `--json-summary` and will not look into some point in the code trying to understand the output of `--json-summary`. It should already be clear from the output. When comments are required, the key names are not good enough in the first place.